#!/usr/bin/python
# Line Processing Configuration

PACKAGE="line_processing"
import roslib; roslib.load_manifest(PACKAGE)

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
#        Name                   Type        Reconfiguration level           Description                                                                     Default     Min Max
gen.add("stock_image_enabled",  bool_t,     SensorLevels.RECONFIGURE_STOP,  "Enables or disables use of a stock image.",                                    False)
gen.add("stock_image",          str_t,      SensorLevels.RECONFIGURE_STOP,  "Location of the stock image.",                                                 "/tmp/stock1.jpg")
gen.add("brightness_threshold", int_t,      SensorLevels.RECONFIGURE_STOP,  "Any pixels with a value less than this is zero'ed",                            245,        0,  255)
gen.add("histogram_enabled",    bool_t,     SensorLevels.RECONFIGURE_STOP,  "Enables of disables the use of the Histogram",                                 True)
gen.add("hough1_min_points",    int_t,      SensorLevels.RECONFIGURE_STOP,  "The minimum number of colinear points needed to be considered a line.",        10,         0,  200)
gen.add("hough1_max_spacing",   int_t,      SensorLevels.RECONFIGURE_STOP,  "The maximum pixels between lines before splitting them.",                      10,          0,  100)
gen.add("hough2_enabled",       bool_t,     SensorLevels.RECONFIGURE_STOP,  "Enables of disables the use of the Secondary Hough Transform",                 True)
gen.add("hough2_min_points",    int_t,      SensorLevels.RECONFIGURE_STOP,  "The minimum number of colinear points needed to be considered a line.",        105,         0,  200)
gen.add("hough2_max_spacing",   int_t,      SensorLevels.RECONFIGURE_STOP,  "The maximum pixels between lines before splitting them.",                      10,          0,  100)

exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "line_processing"))